packages/core/content-releases/admin/src/components/ReleaseModal.tsx

```diff
@@ -34,6 +49,35 @@ export const ReleaseModal = ({
   const { formatMessage } = useIntl();
   const { pathname } = useLocation();
   const isCreatingRelease = pathname === `/plugins/${pluginId}`;
+  // Set default first value from the timezone list with offset UTC+00:00
+  const { timezoneList, currentTimezone = { value: 'Africa/Abidjan' } } = useTimezone();
+
+  /**
+   * Generate scheduled time using selected date, time and timezone
+   */
+  const getScheduledTimestamp = (values: FormValues) => {
+    const { date, time, timezone } = values;
+    if (!date || !time || !timezone) return null;
+
+    const [hours, minutes] = time.split(':').map(Number);
+    const combinedDateTime = new Date(date);
+    combinedDateTime.setHours(hours);
+    combinedDateTime.setMinutes(minutes);
+    const selectedTimezone = timezoneList.find((zone) => zone.value === timezone);
+    const timezoneOffset = selectedTimezone!.offset.slice(3);
+    const offset = timezoneOffset.length > 0 ? parseInt(timezoneOffset, 10) : 0;
+
+    // Adjust the date based on the selected timezone offset
+    const combinedTime = combinedDateTime.getTime();
+    const combinedTimeWithOffset = combinedTime - offset * 60 * 60 * 1000; // Convert offset in hours to milliseconds
+
+    // Because new Date always adds local timezone offset, remove it to set the correct UTC time
+    const dateObject = new Date(combinedTimeWithOffset);
+    const scheduledDate = new Date(
+      dateObject.getTime() - dateObject.getTimezoneOffset() * 60 * 1000
+    );
+    return scheduledDate;
+  };
```

Comment:
I think you should be doing something similar to how we do the TimePicker in the DS for string manipulation, this is prone to bugs and the separator isn't guaranteed to be `:` because it comes from the web APIs stuff:

https://github.com/strapi/design-system/blob/8f6f2cf7d047e6cad4b17270f1950fd9426e29d7/packages/strapi-design-system/src/TimePicker/TimePicker.tsx#L88